### PR TITLE
fix(state): `CoreAccessor` manages its own context

### PR DESCRIPTION
**Pull request by renaynay**
Resolves https://github.com/celestiaorg/celestia-node/issues/1248 and https://github.com/celestiaorg/celestia-node/issues/1246

### CODE REVIEW

**Review by Wondertan — DISMISSED** [review]
For this to be tested. We need either a Swamp test that sends PFDs or a per-module API testing. The latter would construct the API/Module, and test each method on the API

File: state/core_access.go [hunk 1]
```diff
@@ -100,16 +101,25 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
...
+		log.Warn("core accessor already stopped")
+		return nil
+	}
 	if ca.coreConn == nil {
 		log.Warn("no connection found to close")
 		return nil
 	}
+	defer func() {
+		ca.cancel()
+		ca.cancel = nil
```
Comment (MSevey) [inline state/core_access.go:114]: [non-blocking][recommendation]
When there are multiple steps to something like this, I think it is nice to have a helper function to ensure that, if we need to call cancel in other places, we don't miss a step.
```go
func (ca *CoreAccessor) cancelCTX() {
    ca.cancel()
    ca.cancel = nil
}
```

**Review by distractedm1nd — APPROVED** [review]

File: state/core_access.go (hunk 1 above)
Comment (renaynay) [reply state/core_access.go]: @MSevey done.

**Review by Wondertan — DISMISSED** [review]

**Comment by renaynay** [comment]
> For this to be tested. We need either a Swamp test that sends PFDs or a per-module API testing. The latter would construct the API/Module, and test each method on the API

we have testground tests that incorporate PFD submission cc @Bidon15

**Comment by Wondertan** [comment]
> we have testground tests that incorporate PFD submission cc @Bidon15

True, the problem is that it didn't catch it

**Comment by renaynay** [comment]
@Wondertan I think they *did* catch it because that's how @Bidon15 found the error

**Comment by renaynay** [comment]
Actually it was found via testing directly on arabica, I stand corrected

File: state/core_access.go [hunk 2]
```diff
@@ -100,21 +101,32 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
 }
 
 func (ca *CoreAccessor) Stop(context.Context) error {
-	defer ca.cancel()
+	if ca.cancel == nil {
```
Comment (MSevey) [inline state/core_access.go:104]: a new condition should result in a new/updated test :-)

**Review by Bidon15 — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**